app/helpers/test_executions_results_helper.rb

```diff
@@ -0,0 +1,69 @@
+module TestExecutionsResultsHelper
+  def get_select_history_message(execution, is_most_recent)
+    msg = ''
+    msg << 'Most Recent - ' if is_most_recent
+    msg << execution.created_at.in_time_zone('Eastern Time (US & Canada)').strftime('%b %d, %Y at %I:%M %p (%A)')
+    msg << ' (passing)' if execution.passing?
+    msg << ' (in progress)' if execution.incomplete?
+    if execution.failing?
+      num_errors = execution.execution_errors.count
+      num_errors += execution.sibling_execution.execution_errors.count if execution.sibling_execution
+      msg << " (#{num_errors} errors)"
+    end
+    msg
+  end
+
+  # # # # # # # # # # # # # # # #
+  #   E r r o r   T a b l e s   #
+  # # # # # # # # # # # # # # # #
+
+  def supplemental_data_errors(errors)
+    return_errors = errors.select do |err|
+      err.has_attribute?('error_details') && err['error_details'].key?('type') && err['error_details']['type'] == 'supplemental_data'
+    end
+    # sort by population key (IPP, DENOM, ...)
+    return_errors.sort do |a, b|
+      a.error_details.population_key <=> a.error_details.population_key
+    end
+  end
```

Comment:
this is the only function I added. I spit the test_executions helper class because the file was over 100 lines